docs/template-functions.md

```diff
@@ -0,0 +1,199 @@
+
+```
+{{ render_navigation(pages.homepage.navigation) }}
+```
+
+Which would produce an output like this:
+
+```
+<ul>
+  <li>
+    <a href="/page-1/">Page One</a>
+  </li>
+
+  <li>
+    <a href="/page-2/">Page Two</a>
+  </li>
+</ul>
+```
+
+If you would like the "base page" (the page that the navigation is being based off) to be included in the navigation simply add the `section` kwarg:
+
+```
+{{ render_navigation(pages.homepage.navigation, section=pages.homepage) }}
+```
+
+The output of this would be:
+
+```
+<ul>
+  <li>
+    <a class="here" href="/">Homepage</a>
+  </li>
+
+  <li>
+    <a href="/page-1/">Page One</a>
+  </li>
+
+  <li>
+    <a href="/page-2/">Page Two</a>
+  </li>
+</ul>
+```
+
+You will probably want to override the output in your project. For example, you'll almost certainly want to add class names, and to handle children of children. We suggest you make a local copy of `apps/pages/templates/pages/navigation.html` in your project.
+
+### `render_breadcrumbs(page=None, extended=False)`
+
+Renders the breadcrumbs trail for the current page:
+
+```
+{{ render_breadcrumbs() }}
+```
+
+To override and extend the breadcrumb trail within page applications, add the `extended` flag to the tag and add your own breadcrumbs underneath:
+
+```
+{{ render_breadcrumbs(extended=1) }}
+```
+
+### `get_page_url(page, view_func=None, *args, **kwargs)`
+
+Resolves the URL of a route defined in a page's `urlconf`, passing positional and/or keyword arguments to the resolver.
+It is a thin wrapper around `Page.reverse`.
+
+```
+{{ get_page_url(pages.current, 'article_detail', slug=article.slug) }}
+```
+
+### `get_canonical_url()`
+
+Returns the canonical URL for the currently viewed URL.
+It merely ensures that any query string junk does not cause a page to be indexed more than once by search engines.
+
+## OpenGraph (Facebook and others) functions
+
+`get_og_title`, `get_og_description` and `get_og_image` render the OpenGraph title, description, and image for the current object.
+
+You will want to use them like so:
+
+```
+<meta property="og:title" content="{{ get_og_title() }}">
+<meta property="og:url" content="{{ get_canonical_url() }}">
+<meta property="og:type" content="website">
+<meta property="og:description" content="{{ get_og_description() }}">
+<meta property="og:image" content="{{ get_og_image() }}">
+```
+
+`get_og_title` is smart enough to fall back to the current object's browser title override or `title` attribute.
+
+`get_og_image` checks the OpenGraph image field of the current object, then falls back to the `image`, `photo` and `logo` field on the current object, in that order.
+
+`get_og_description` will fall back to the current object's `description` and `summary` field.
+
+## Twitter card functions
+
+`get_twitter_card`, `get_twitter_description` and `get_twitter_image` render the current Twitter card information for the current object. Usage:
```

Comment:
A stylistic point more than anything - Previously you wrote "You will want to use them like so:" on a new line.